test/addons-napi/test_reference/test.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+// Flags: --expose-gc
+
+const common = require('../../common');
+const assert = require('assert');
+
+const test_reference = require(`./build/${common.buildType}/test_reference`);
+
+// This test script uses external values with finalizer callbacks
+// in order to track when values get garbage-collected. Each invocation
+// of a finalizer callback increments the finalizeCount property.
+assert.strictEqual(0, test_reference.finalizeCount);
+
+// External value without a finalizer
+let value = test_reference.createExternal();
```

Comment:
Can you turn these groups into `{` … `}` blocks and use `const` where that works? That makes it a bit easier to see how different parts of a test (don’t) interact :)